### ISSUE TITLE
Error E0718 is not an actual error

### ISSUE THREAD

**Issue by Danny-06** [quick-lint/quick-lint-js#1199]
The error as documented in the official docs [https://quick-lint-js.com/errors/E0718/](https://quick-lint-js.com/errors/E0718/)
implies that the following scenario should throw an error:
```js
const bug = { milestone: null };
console.log(bug.milestone);               // null
console.log(bug.milestone?.name);         // undefined
console.log(bug.milestone?.name.trim());  // throws an error
```
But the actual output when running the code is:
```js
const bug = { milestone: null };
console.log(bug.milestone);               // null
console.log(bug.milestone?.name);         // undefined
console.log(bug.milestone?.name.trim());  // undefined
```

Is there a good reason for this error to be listed or should be removed?

**Comment by strager** [comment]
> Is there a good reason for this error to be listed or should be removed?

This diagnostic should be removed. In fact, it is removed in master (#1192). I will ship the fix in version 3.2.

**Comment by tomocrafter** [comment]
@strager Thank you!
now I wonder why it is introduced first, like was it actually occurs the error?

**Comment by strager** [comment]
>  now I wonder why it is introduced first, like was it actually occurs the error?

Nah, my brain just wasn't working. 😄

**Comment by strager** [comment]
Fixed released in [version 3.2.0](https://quick-lint-js.com/releases/#3.2.0).
